### PR TITLE
20241212 Jesse Meng add Utah SLR 2024 algorithm

### LINKED ISSUE #1101: Transition the updated U Utah SLR algorithm to UPP
U Utah has updated the snow-liquid-ration (SLR) algorithm trained with data across CONUS (including west and east). The new algorithm will be transitioned to UPP.

### PR DESCRIPTION
**Pull request by JesseMeng-NOAA**
Add Utah SLR 2024 algorithm.

### CODE REVIEW

File: sorc/ncep_post.fd/UPP_PHYSICS.f [hunk 1]
```diff
@@ -4497,6 +4500,222 @@ SUBROUTINE CALSLR_UUTAH(SLR)
...
+!> @param[out] SLR real Solid snow to liquid ratio
+!> 
+!> ### Program history log:
+!> Date | Programmer | Comments
+!> -----|------------|---------
+!> 2024-11-15 | Jesse Meng | Initial
+!>
+!> @author Jesse Meng @date 2024-11-15
+
+      SUBROUTINE CALSLR_UUTAH2(SLR)
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/UPP_PHYSICS.f]: @JesseMeng-NOAA Can you update this new subroutine to follow the format below?

- all lowercase

- indentation for loops, conditionals and blocks

**Comment by WenMeng-NOAA** [comment]
@JesseMeng-NOAA From the UPP RTs on Hera, your PR impacts on the field "SDEN" in the tests for gfs, fv3r and 3drtma. I assume these are expected changes. Please take a look at /home/Wen.Meng/stmp2/UPP-RTs/fv3gfs_2019083000 and /home/Wen.Meng/stmp2/UPP-RTs/fv3r_2023062800 and let me know if you see issues.

**Comment by JesseMeng-NOAA** [comment]
> @JesseMeng-NOAA From the UPP RTs on Hera, your PR impacts on the field "SDEN" in the tests for gfs, fv3r and 3drtma. I assume these are expected changes. Please take a look at /home/Wen.Meng/stmp2/UPP-RTs/fv3gfs_2019083000 and /home/Wen.Meng/stmp2/UPP-RTs/fv3r_2023062800 and let me know if you see issues.

Results look reasonable. Thanks.

**Comment by WenMeng-NOAA** [comment]
@JesseMeng-NOAA Please add a change log in MDL2P.f

**Comment by JesseMeng-NOAA** [comment]
> @JesseMeng-NOAA Please add a change log in MDL2P.f

Added.

**Comment by WenMeng-NOAA** [comment]
@gspetro-NOAA You may start the UPP RTs on R&D machines. Please refer my tests on Hera:
```
Wen.Meng@hfe03 UPP-RTs$ pwd
/home/Wen.Meng/stmp2/UPP-RTs
Wen.Meng@hfe03 UPP-RTs$ ls -ltr */*diff
-rw-r--r-- 1 Wen.Meng stmp    75 Dec 17 16:27 rap_2020072316_pe_test/WRFPRS.GrbF16.diff
-rw-r--r-- 1 Wen.Meng stmp 14011 Dec 17 16:29 rtma_2023040400_pe_test/NATLEV00.tm00.diff
-rw-r--r-- 1 Wen.Meng stmp 13861 Dec 17 16:30 fv3r_2023062800/NATLEV10.tm00.diff
-rw-r--r-- 1 Wen.Meng stmp 13861 Dec 17 16:30 fv3r_2023062800_pe_test/NATLEV10.tm00.diff
-rw-r--r-- 1 Wen.Meng stmp 14011 Dec 17 16:32 rtma_2023040400/NATLEV00.tm00.diff
-rw-r--r-- 1 Wen.Meng stmp  3449 Dec 17 16:35 fv3gfs_2019083000/gfs.t00z.master.grb2f006.diff
-rw-r--r-- 1 Wen.Meng stmp  3449 Dec 17 16:36 fv3gfs_2019083000_pe_test/gfs.t00z.master.grb2f006.diff
```
I will conduct the RTs on WCOSS2 next week when WCOSS2 is back.

**Comment by WenMeng-NOAA** [comment]
The UPP RTs have been completed on WCOSS2. The baseline recreations are needed for gfs, fv3r and 3drtma with this PR.

**Comment by gspetro-NOAA** [comment]
Tests pass on Hera, Orion, and Hercules with the expected baseline changes.

**Comment by WenMeng-NOAA** [comment]
@gspetro-NOAA Please approve this PR.

**Review by WenMeng-NOAA — APPROVED** [review]

**Comment by WenMeng-NOAA** [comment]
This PR is ready for merging.

**Pull request merged**